client/app/bundles/course/duplication/pages/Duplication/DestinationCourseSelector/index.tsx

```diff
@@ -0,0 +1,120 @@
+    id: 'course.duplication.Duplication.DestinationCourseSelector.selectDestinationCoursePrompt',
+    defaultMessage: 'Select destination course:',
+  },
+  defaultTitle: {
+    id: 'course.duplication.Duplication.DestinationCourseSelector.defaultTitle',
+    defaultMessage: '{title} (Copied at {timestamp})',
+  },
+  success: {
+    id: 'course.duplication.Duplication.DestinationCourseSelector.success',
+    defaultMessage: 'Duplication is successful. Redirecting to the new course.',
+  },
+  pending: {
+    id: 'course.duplication.Duplication.DestinationCourseSelector.pending',
+    defaultMessage:
+      'Please wait as your request to duplicate the course is being processed.\n\
+    You may close the window while duplication is in progress and\n\
+    you will also receive an email with a link to the new course when it becomes available.',
+  },
+  failure: {
+    id: 'course.duplication.Duplication.DestinationCourseSelector.failure',
+    defaultMessage: 'Duplication failed.',
+  },
+});
+
+const DestinationCourseSelector: FC = () => {
+  const { t } = useTranslation();
+  const dispatch = useAppDispatch();
+
+  const duplication = useAppSelector(selectDuplicationStore);
+  const {
+    destinationInstanceId,
+    destinationCourses,
+    destinationCourseTitle,
+    destinationCourseStartAt,
+    currentHost,
+    currentCourseId,
+    destinationCourseId,
+    duplicationMode,
+    sourceCourse,
+    metadata: { currentInstanceId },
+    destinationInstances,
+    isDuplicating,
+  } = duplication;
+
+  const NewCourseDestinationForm = (): JSX.Element => {
+    const tomorrow = moment().add(1, 'day');
+    const defaultNewCourseStartAt = moment(sourceCourse.start_at).set({
+      year: tomorrow.year(),
+      month: tomorrow.month(),
+      date: tomorrow.date(),
+    });
+
+    const timeNow = moment().format(SHORT_DATE_TIME_FORMAT);
+    const newTitleValues = { title: sourceCourse.title, timestamp: timeNow };
+    const initialValues = {
+      destination_instance_id: destinationInstanceId ?? currentInstanceId,
+      new_title:
+        destinationCourseTitle ?? t(translations.defaultTitle, newTitleValues),
+      new_start_at: destinationCourseStartAt
+        ? moment(destinationCourseStartAt)
+        : defaultNewCourseStartAt,
+    };
+
+    return (
+      <NewCourseForm
+        disabled={isDuplicating}
+        initialValues={initialValues}
+        onSubmit={(values, setError) =>
+          dispatch(
+            duplicateCourse(
+              values,
+              destinationInstances[values.destination_instance_id]?.host,
+              t(translations.success),
+              t(translations.pending),
+              t(translations.failure),
+              setError,
+            ),
+          )
+        }
+      />
+    );
+  };
+
+  const ExistingCourseDestinationForm = (): JSX.Element => {
+    return (
+      <CourseDropdownMenu
+        additionalClassName="destination-course-dropdown"
```

Comment:
Do you really need to create an additional prop for this? I don't see `CourseDropdownMenu` used anywhere else.

Reply:
in the previous version (while still using jsx) this is how they implement it. And after some contemplation, this is more sensible than combining all since this additional class name is only for the testing case (rspec, when they want to trigger the clicking of the dropdown menu) and hence having different purpose than the className defined inside this CourseDropdownMenu

By meaning, let's say we want to create another component that needs this, inside rspec test later, we can differentiate easier between this component and the later-defined component

Reply:
In that case, your `onChange` function on the `<Select>` component should also remain on this level.  
Currently the dispatch is specific to `actions.setDestinationCourseId` instead of being agnostic.

In terms of rspec, we can alternatively give this `<CourseDropdownMenu className="destination-course-dropdown-menu">` and then use `.find('.destination-course-dropdown select')` which will find the descendant <select> tag.